src/components/Schedule/CalendarView.tsx

```diff
@@ -72,43 +17,85 @@ const getWeekDay = (day: Date) => {
 };
 
 //course assignments that are flagged to be on multiple days should be set for those days
-const splitCourseDays = (assignment: Assignment) => {
-  const { monday, tuesday, wednesday, thursday, friday } = assignment;
+const splitCourseDays = (course: Course) => {
+  const { monday, tuesday, wednesday, thursday, friday, beginTime, endTime } =
+    course.meetingTime;
   const daysArray = [monday, tuesday, wednesday, thursday, friday];
 
+  //2022-05-31T is a monday, and we just need the base to be a monday
+  //HH:MM:SS.000-7:00 for time format, it comes in HHMM format.
+  const beginHoursMinutes = [
+    beginTime.slice(0, beginTime.length / 2),
+    beginTime.slice(-beginTime.length / 2),
+  ];
+  const endHourMinutes = [
+    endTime.slice(0, endTime.length / 2),
+    endTime.slice(-endTime.length / 2),
+  ];
+
+  const begin =
+    "2022-05-31T" +
+    beginHoursMinutes[0] +
+    ":" +
+    beginHoursMinutes[1] +
+    ":00.000-07:00";
+  const end =
+    "2022-05-31T" +
+    endHourMinutes[0] +
+    ":" +
+    endHourMinutes[1] +
+    ":00.000-07:00";
+
   const splitAppointments = daysArray.map((day, index) => {
-    if(!day){
+    if (!day) {
       return null;
     }
-    //2022-05-31T is a monday, and we just need the base to be a monday
-    const begin = "2022-05-31T" + assignment.beginTime;
-    const end = "2022-05-31T" + assignment.endTime;
+
     const startDate = new Date(begin);
     const endDate = new Date(end);
 
     //change the date based on what day of the week it's supposed to be
     startDate.setDate(startDate.getDate() + index - 1);
     endDate.setDate(endDate.getDate() + index - 1);
 
-    const appointment = {
-      ...assignment,
+    const { meetingTime, ...appointment } = {
+      ...course,
       startDate: startDate,
       endDate: endDate,
     };
     return appointment;
   });
-  const validAppointments = splitAppointments.filter((appointment) => appointment !== null) as Assignment[];
+  const validAppointments = splitAppointments.filter(
+    (appointment) => appointment !== null
+  ) as Appointment[];
   return validAppointments;
 };
 
-export const CalendarView = () => {
+//convert Course object data from backend to the structure DevExtreme expects
+const buildAppointments = (courses: Course[]) => {
+  const appointments = courses
+    .map((course) => {
+      const assignedCourses = splitCourseDays(course);
+      return assignedCourses;
+    })
+    .flat();
```

Comment:
Nitpick, but you can just use `flatMap` here

Reply:
oh  that's nice, I didn't even know that existed lol